<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function takes harmonized exposure/outcome data and runs Mendelian Randomization Bayesian Model Averaging to identify and prioritize relationships between the exposures and outcome."><title>Perform Mendelian Randomization Bayesian Model Averaging — mr_bma • mrbma</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Perform Mendelian Randomization Bayesian Model Averaging — mr_bma"><meta property="og:description" content="This function takes harmonized exposure/outcome data and runs Mendelian Randomization Bayesian Model Averaging to identify and prioritize relationships between the exposures and outcome."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mrbma</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/get-started.html">Get started</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mglev1n/mrbma/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Perform Mendelian Randomization Bayesian Model Averaging</h1>
      <small class="dont-index">Source: <a href="https://github.com/mglev1n/mrbma/blob/HEAD/R/mr_bma.R" class="external-link"><code>R/mr_bma.R</code></a></small>
      <div class="d-none name"><code>mr_bma.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes harmonized exposure/outcome data and runs Mendelian Randomization Bayesian Model Averaging to identify and prioritize relationships between the exposures and outcome.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mr_bma</span><span class="op">(</span></span>
<span>  <span class="va">harmonized_data</span>,</span>
<span>  prior_prob <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  prior_sigma <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  top <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  <span class="va">kmin</span>,</span>
<span>  <span class="va">kmax</span>,</span>
<span>  remove_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_influential <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  calculate_p <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nrepeat <span class="op">=</span> <span class="fl">1e+05</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>harmonized_data</dt>
<dd><p>(list) Harmonized data of exposures/outcomes of interest, generated using <code><a href="https://rdrr.io/pkg/TwoSampleMR/man/mv_harmonise_data.html" class="external-link">TwoSampleMR::mv_harmonise_data()</a></code></p></dd>


<dt>prior_prob</dt>
<dd><p>(numeric) Prior probability</p></dd>


<dt>prior_sigma</dt>
<dd><p>(numeric) Variance of the prior probability</p></dd>


<dt>top</dt>
<dd><p>(numeric) Number of top models to return in results</p></dd>


<dt>kmin</dt>
<dd><p>(numeric) Minimum model size. By default the model will consider all combinations of exposures (eg. <code>kmin</code> = <code>kmax</code>), but may become computationally infeasible with &gt;12 exposures. If <code>kmin</code> &lt; <code>kmax</code>, then a stochastic search is performed.</p></dd>


<dt>kmax</dt>
<dd><p>(numeric) Maximum model size. By default the model will consider all combinations of exposures (eg. <code>kmin</code> = <code>kmax</code>), but may become computationally infeasible with &gt;12 exposures. If <code>kmin</code> &lt; <code>kmax</code>, then a stochastic search is performed.</p></dd>


<dt>remove_outliers</dt>
<dd><p>(logical) Remove outlier variants based on Q-statistic</p></dd>


<dt>remove_influential</dt>
<dd><p>(logical) Remove influential variants based on Cook's distance</p></dd>


<dt>calculate_p</dt>
<dd><p>(logical) Use empirical permutation with Nyholt procedure of effective tests to estimate p-values</p></dd>


<dt>nrepeat</dt>
<dd><p>(numeric) Number of permutations used when estimating Nyholt-corrected False Discovery Rate</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code>list</code> containing the results of MR-BMA analyses. The list includes:</p><ul><li><p><code>model_best</code> = A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> containing a list of the top models</p></li>
<li><p><code>mip_table</code> = A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> containing the marginal inclusion probabilities of each risk factor</p></li>
<li><p><code>mrbma_output</code> = Raw output from the <code>summarymvMR_SSS</code> function</p></li>
<li><p><code>influential_res</code> = Diagnostic plots representing the detection of influential variants</p></li>
<li><p><code>outlier_res</code> = Diagnostic plot representing the detection of outlier variants</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Extract genetic instruments</span></span></span>
<span class="r-in"><span><span class="va">lipid_exposures</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/mv_extract_exposures.html" class="external-link">mv_extract_exposures</a></span><span class="op">(</span>id_exposure <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-a-299"</span>, <span class="st">"ieu-a-300"</span>, <span class="st">"ieu-a-302"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Clumping 1, 214 variants, using EUR population reference</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing 70 of 214 variants due to LD with other variants or absence from LD reference panel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 144 SNP(s) from 3 GWAS(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising HDL cholesterol || id:ieu-a-299 (ieu-a-299) and LDL cholesterol || id:ieu-a-300 (ieu-a-300)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising HDL cholesterol || id:ieu-a-299 (ieu-a-299) and Triglycerides || id:ieu-a-302 (ieu-a-302)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Extract corresponding outcome data</span></span></span>
<span class="r-in"><span><span class="va">cad_outcome</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html" class="external-link">extract_outcome_data</a></span><span class="op">(</span>snps <span class="op">=</span> <span class="va">lipid_exposures</span><span class="op">$</span><span class="va">SNP</span>, outcomes <span class="op">=</span> <span class="st">"ebi-a-GCST005195"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 144 SNP(s) from 1 GWAS(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finding proxies for 2 SNPs in outcome ebi-a-GCST005195</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 2 SNP(s) from 1 GWAS(s)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Generate harmonized dataset</span></span></span>
<span class="r-in"><span><span class="va">lipids_cad_harmonized</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/mv_harmonise_data.html" class="external-link">mv_harmonise_data</a></span><span class="op">(</span><span class="va">lipid_exposures</span>, <span class="va">cad_outcome</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising HDL cholesterol || id:ieu-a-299 (ieu-a-299) and Coronary artery disease || id:ebi-a-GCST005195 (ebi-a-GCST005195)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> rs7534572</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run MR-BMA</span></span></span>
<span class="r-in"><span><span class="va">mr_bma_res</span> <span class="op">&lt;-</span> <span class="fu">mr_bma</span><span class="op">(</span><span class="va">lipids_cad_harmonized</span>, calculate_p <span class="op">=</span> <span class="cn">TRUE</span>, nrepeat <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Preparing input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Preparing input <span style="color: #B2B2B2;">[78ms]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Running MR-BMA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Running MR-BMA <span style="color: #B2B2B2;">[45ms]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Removing influential variants</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Removing influential variants <span style="color: #B2B2B2;">[89ms]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Removing outliers</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Removing outliers <span style="color: #B2B2B2;">[95ms]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Estimating empirical p-value using 1000 permutations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Estimating empirical p-value using 1000 permutations <span style="color: #B2B2B2;">[2.8s]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Performing Nyholt correction for effective number of tests</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Performing Nyholt correction for effective number of tests <span style="color: #B2B2B2;">[28ms]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Output best models</span></span></span>
<span class="r-in"><span><span class="va">mr_bma_res</span><span class="op">$</span><span class="va">model_best</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 7 × 3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   rf_combination                posterior_prob causal_estimate  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> ieu-a-300,ieu-a-302                    0.843 0.365,0.219      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> ieu-a-299,ieu-a-300,ieu-a-302          0.156 -0.063,0.361,0.19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> ieu-a-299,ieu-a-300                    0     -0.131,0.376     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> ieu-a-300                              0     0.393            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> ieu-a-302                              0     0.311            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> ieu-a-299,ieu-a-302                    0     -0.105,0.26      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">7</span> ieu-a-299                              0     -0.202           </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Output marginal inclusion probabilities for each risk factor</span></span></span>
<span class="r-in"><span><span class="va">mr_bma_res</span><span class="op">$</span><span class="va">mip_table</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 3 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   exposure  mace                 mip               p_val           p_fdr p_nyh…¹</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> ieu-a-299 -0.00987190503160796 0.15651065273458  0.935064935064… 0.93… 1      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> ieu-a-300 0.364151410155237    1                 0.000999000999… 0.00… 0.0019…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> ieu-a-302 0.214705609094214    0.999726011149142 0.000999000999… 0.00… 0.0019…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with abbreviated variable name ¹​p_nyholt</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Michael Levin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

